File: csharp/lucians-luscious-lasagna/LuciansLusciousLasagna.cs
class Lasagna
{
    public int ExpectedMinutesInOven() => 40;

    public int RemainingMinutesInOven(int timeIn) => 
        this.ExpectedMinutesInOven() - timeIn;

    public int PreparationTimeInMinutes(int layers) => layers * 2;

    public int ElapsedTimeInMinutes(int layers, int timeIn) => 
        this.PreparationTimeInMinutes(layers) + timeIn;
}

File: fsharp/lucians-luscious-lasagna/LuciansLusciousLasagna.fs
module LuciansLusciousLasagna

let expectedMinutesInOven = 40

let remainingMinutesInOven minutesIn = expectedMinutesInOven - minutesIn

let preparationTimeInMinutes layers = layers * 2

let elapsedTimeInMinutes layers minutesIn = 
    let layerTime = preparationTimeInMinutes layers
    layerTime + minutesIn
